lib/uklock/include/uk/mutex.h

```diff
@@ -90,61 +97,68 @@ extern __spinlock              _uk_mutex_metrics_lock;
 #endif /* CONFIG_LIBUKLOCK_MUTEX_METRICS */
 
 #define	UK_MUTEX_INITIALIZER(name)				\
-	{ 0, NULL, __WAIT_QUEUE_INITIALIZER((name).wait) }
+	{ 0, 0, NULL, __WAIT_QUEUE_INITIALIZER((name).wait) }
 
-void uk_mutex_init(struct uk_mutex *m);
+void __uk_mutex_init(struct uk_mutex *m, uint8_t config_flags);
 void uk_mutex_get_metrics(struct uk_mutex_metrics *dst);
 
+#define uk_mutex_init(m) __uk_mutex_init(m, 0)
+#define uk_mutex_init_config(m, config_flags) __uk_mutex_init(m, config_flags)
+
 static inline void uk_mutex_lock(struct uk_mutex *m)
 {
 	struct uk_thread *current;
-	unsigned long irqf;
 
 	UK_ASSERT(m);
 
 	current = uk_thread_current();
 
+	/* If the owner is a current thread, then we increment the lock count. */
+	if (__IS_CONFIG_WRITE_RECURSE(m->config_flags)
+			&& unlikely(m->owner == current)) {
```

Comment:
I think it is safe to assume that recursive locks are less used. So if you want to use the `unlikely` attribute, I would put it around the whole condition.
```suggestion
	if (unlikely(uk_mutex_is_recursive(m) && m->owner == cur)) {
```